File: cpp/questions/repeatedNTimes.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        std::map<int, int> numcnt;
        int len = nums.size() / 2;
        for (int& num : nums) {
            if (++numcnt[num] == len) {
                return num;
            }
        }
        return -1;
    }
};

File: python/questions/repeatedNTimes.py
class Solution:
    def repeatedNTimes(self, nums: List[int]) -> int:
        numcnt = {}
        n = len(nums) // 2
        for num in nums:
            numcnt[num] = numcnt.get(num, 0) + 1
            if numcnt[num] == n:
                return num
        return -1
